### ISSUE TITLE
Can't override DataGrid ColumnHeadersAndRowsSeparator Rectangle's Height

### ISSUE THREAD

**Issue by Saliom** [AvaloniaUI/Avalonia#14520]
## Describe the bug
It is not possible to override locally set Height value for `Avalonia.Controls.DataGrid /template/ Rectangle#PART_ColumnHeadersAndRowsSeparator` that is set here:
https://github.com/AvaloniaUI/Avalonia/blob/c2e9679faeae30dda888c47a994d2b0180cb7004/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L530-L534

## To Reproduce

1. Recreate the first example of the docs at https://docs.avaloniaui.net/docs/reference/controls/datagrid/
2. Add the following `Style`:
```XAML
<Style Selector="DataGrid /template/ Rectangle#PART_ColumnHeadersAndRowsSeparator">
	<Setter Property="Height" Value="20"/>
</Style>
```
3. Start application

## Expected behavior
The ColumnHeadersAndRowsSeparator should have its Height set to 20.

## Screenshots
![image](https://github.com/AvaloniaUI/Avalonia/assets/7543712/f3d67faf-9662-4efd-bce3-2418436595ef)

## Environment
- OS: Windows 10 21H2
- Avalonia-Version: 11.0.7

**Comment by maxkatz6** [comment]
Alternatively, you can replace template with a style.

**Comment by Saliom** [comment]
> Alternatively, you can replace template with a style.

Could you please provide more detailed steps or an example of replacing a template with a style.

**Comment by timunie** [comment]
@Saliom best to looking up DataGridStyle in source and cooy the relevant parts.

**Comment by timunie** [comment]
> Could you please provide more detailed steps or an example of replacing a template with a style?

Given you use the SimpleTheme, it is this setter to copy: https://github.com/AvaloniaUI/Avalonia/blob/0c0d4497ef156aa6defaa4cce05fac5831576dd4/src/Avalonia.Controls.DataGrid/Themes/Simple.xaml#L313-L372

**Comment by timunie** [comment]
Yeah adding another DynamicResource sounds like a good idea if anyone wants to file a PR for it.
